lib/pbench/common/logger.py

```diff
@@ -81,17 +81,21 @@ class _PbenchLogFormatter(logging.Formatter):
     def __init__(self, fmt=None, datefmt=None, style="{", max_line_length=0):
         super().__init__(fmt=fmt, datefmt=datefmt, style=style)
         self.max_line_length = max_line_length
+        self.converter = datetime.utcfromtimestamp
+
+    def formatTime(self, record, datefmt=None):
+        """
+        Return the creation time of the specified LogRecord as formatted text.
+        """
+        return self.converter(record.created).isoformat()
```

Comment:
This is a bit interesting as we're still bypassing the `datefmt` setting and forcing `isoformat()`. I'm not objecting, since I think that's what we want to see... but I I'm also not sure this is really any cleaner than simply doing the formatting inline in `format()`. Also, is the default `converter` supposed to do UTC, or are we now generating local time? The documentation isn't entirely clear on that point; but, again, this isn't necessarily equivalent to what we had before. Should we be overriding `converter` also, as I mentioned in #2003??

Reply:
> This is a bit interesting as we're still bypassing the `datefmt` setting and forcing `isoformat()`. I'm not objecting, since I think that's what we want to see... but I I'm also not sure this is really any cleaner than simply doing the formatting inline in `format()`.

It is not cleaner or simpler, but it is quite comparable.  If you were to diff the two routines side-by-side, the differences are in the newline character handling, adding the `.replace()` calls, and in the trimming of the maximum message size.

> Also, is the default `converter` supposed to do UTC, or are we now generating local time? The documentation isn't entirely clear on that point; but, again, this isn't necessarily equivalent to what we had before. Should we be overriding `converter` also, as I mentioned in #2003??

I am not sure I follow these questions.  Aren't we now overriding the `converter` at line 84?

Reply:
> > This is a bit interesting as we're still bypassing the `datefmt` setting and forcing `isoformat()`. I'm not objecting, since I think that's what we want to see... but I I'm also not sure this is really any cleaner than simply doing the formatting inline in `format()`.
> 
> It is not cleaner or simpler, but it is quite comparable. If you were to diff the two routines side-by-side, the > I am not sure I follow these questions. Aren't we now overriding the `converter` at line 84?

Well, ouch; I apparently completely missed that line when I was reading it through!

OK, so we've got our UTC; we're still ignoring `datefmt` and forcing ISO time format. It's less general, but I'm not particularly concerned about that. I'm satisfied.